### DISCUSSION TITLE
tmux and XDG_CONFIG_HOME

### DISCUSSION THREAD

**Discussion by jfindlay** [tmux-python/tmuxp#654] [Q&A] [answered]
Is it possible for tmuxp to work with a tmux config in a "nonstandard" location?  I like to keep my configs in `XDG_CONFIG_HOME`, which tmux does not seem to support (srsly).  Aliasing `tmux` to `tmux -f $XDG_CONFIG_HOME/tmux/config` in the shell means that `tmuxp` does not read the `tmux` config.  Also, creating a wrapper script called `tmux` that basically does the same thing as the alias,
```bash
#!/bin/bash

tmux -f "${XDG_CONFIG_HOME}"/tmux/config "${@}"
```
causes `tmuxp` to infinitely recurse `tmux` subprocesses, whereas tmux itself works fine with this mechanism.

I spent about 15 minutes reading through the source code and I couldn't find where to make any changes for this issue/feature.  With a suggestion, I am willing to implement this myself.

**Comment by tony** [comment, marked as answer]
@jfindlay This is a really good catch. As of 1.6 / 1.7 master:

- `tmuxp` commands (e.g. `tmuxp load`) don't pass through the `-f` (tmux config file) to `tmux`. `tmuxp` already passes through `-2` (256 colors), `-8`, `-L` (socket_name), `-S` (socket_path) to `tmux`

> I spent about 15 minutes reading through the source code and I couldn't find where to make any changes for this issue/feature. With a suggestion, I am willing to implement this myself.

We'd do it in `cli.py` in `command_load` (CLI entrypoint) and `load_workspace`.

We can accept an `-f / --file` as an arg.   We'd need to do this at `t = Server` so the `libtmux` client used refers to the tmux config file:

https://github.com/tmux-python/tmuxp/blob/0d6a0ebc05dcb244b95fa8f8f0735e2a1fd22440/tmuxp/cli.py#L918-L950
https://github.com/tmux-python/tmuxp/blob/0d6a0ebc05dcb244b95fa8f8f0735e2a1fd22440/tmuxp/cli.py#L979-L994
https://github.com/tmux-python/tmuxp/blob/0d6a0ebc05dcb244b95fa8f8f0735e2a1fd22440/tmuxp/cli.py#L575-L584

The good news is, `Server()` accepts a `config_file`: https://github.com/tmux-python/libtmux/blob/6c7c08e718a654f653e0c4388df32e048aa3e445/libtmux/server.py#L79-L94

Do you want to give this a shot @jfindlay? It may be a good first issue

**Reply by jfindlay** [reply]
Cool, yes. I will try to schedule some time in the next few days.

**Reply by tony** [reply]
@jfindlay Do you wanna give this shot? (so you can get credit?)

If not, I can add it and release it (and backport it to 1.6.x and 1.7.x)

**Reply by tony** [reply]
@jfindlay No hurry by the way, just checking to see if you've moved on or will swing back to it. It should be easy for me to add

**Reply by jfindlay** [reply]
Token credit: https://github.com/tmux-python/tmuxp/pull/665.  The real win here though, is getting me to think about tmuxp code which increases the probability I will contribute again in the future.

**Reply by tony** [reply]
@jfindlay 

- https://pypi.python.org/pypi/tmuxp/1.6.5
- https://pypi.python.org/pypi/tmuxp/1.7.1

If you try one of these, how's that?

**Reply by tony** [reply]
> The real win here though, is getting me to think about tmuxp code which increases the probability I will contribute again in the future.

And that's good with me!

**Reply by jfindlay** [reply]
> @jfindlay
> 
>     * https://pypi.python.org/pypi/tmuxp/1.6.5
> 
>     * https://pypi.python.org/pypi/tmuxp/1.7.1
> 
> 
> If you try one of these, how's that?

I have verified the new flag works with both versions.

**Reply by tony** [reply]
Excellent
